### PR TITLE
JBPM-8035: case modeler - correct element properties

### PR DESCRIPTION
**Pull request by xiezhang7**

### CODE REVIEW

**Comment by kiereleaseuser3** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

**Comment by xiezhang7** [comment]
ok to test

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by wmedvede** [comment]
Hi guys @romartin @xiezhang7 @LuboTerifaj 
I'm in a hurry for finishing my own stuff ideally for tomorrow so I'm afraid I won't be able to review this stuff on time :(. I'd suggest guys @romartin @LuboTerifaj do the review and not wait for me.

**Comment by xiezhang7** [comment]
Force pushed to fix merge conflicts.

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by xiezhang7** [comment]
> Hello @xiezhang7
> 
> I cannot test it, because the war hasn't been generated. There is GWT compilation error. Can you have a look at it please?
> 
> Thank you.

Hi @LuboTerifaj , it seems there is something wrong with the Java environment on the machine. Please find the error info below.
Maybe rebuilding it could work, or we will need the administrator's help.
I built it locally yesterday without getting any error.
Thanks.

```
[INFO] #
[INFO] # A fatal error has been detected by the Java Runtime Environment:
[INFO] #
[INFO] #  SIGSEGV (0xb) at pc=0x00007feb7f261b41, pid=11091, tid=0x00007feb41ba0700
[INFO] #
[INFO] # JRE version: Java(TM) SE Runtime Environment (8.0_181-b13) (build 1.8.0_181-b13)
[INFO] # Java VM: Java HotSpot(TM) 64-Bit Server VM (25.181-b13 mixed mode linux-amd64 compressed oops)
[INFO] # Problematic frame:
[INFO] # V  [libjvm.so+0xa38b41]  Symbol::increment_refcount()+0x1
[INFO] #
[INFO] # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
[INFO] #
[INFO] # An error report file with more information is saved as:
[INFO] # /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-downstream-pullrequests/hs_err_pid11091.log
[INFO] #
[INFO] # Compiler replay data is saved as:
[INFO] # /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-downstream-pullrequests/replay_pid11091.log
[INFO] #
[INFO] # If you would like to submit a bug report, please visit:
[INFO] #   http://bugreport.java.com/bugreport/crash.jsp
[INFO] #
```

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

**Comment by xiezhang7** [comment]
Hi @LuboTerifaj , I re-based the pull request to include the latest changes from master branch, and built it locally. There was no build failure for me. I hope the build would succeed this time. Thanks.

**Comment by wmedvede** [comment]
@xiezhang7 @LuboTerifaj what's the status of this guys?

**Comment by xiezhang7** [comment]
> @xiezhang7 @LuboTerifaj what's the status of this guys?

Hi @wmedvede , this pull request is still open. Please feel free to add comments or assign yourself as a reviewer. Thanks.

**Comment by xiezhang7** [comment]
Hi @LuboTerifaj , please find my replies below. Thanks.

> * Shouldn't be "AdHocCompletionCondition" renamed to "Completion Condition" in bpmn process Designer as well? If yes, is it scope of this Jira or not? Because currently it is changed only for Case Management.

@hasys requested this change for case modeler. I am not sure if bpmn process designer should also be updated. 

> * In Jira (https://issues.jboss.org/browse/JBPM-8035) description, there is mentioned, that ad-hoc autostart property should be part of all elements in Case Management. What should happen with these properties for all these elements in case it will be opened in process designer?

I think these properties will be ignored in process designer. Because these properties are optional metadata in the *.bpmn2 files. The marshaller won't read them for process designer.   

> * There is new property which is not described in Jira: "SLA Due date". Shouldn't be there some format checking and also hint, how the value should look like?

This property was introduced in [PR#2308](https://github.com/kiegroup/kie-wb-common/pull/2308/files#diff-c106d4bed2ce52c6d6656d0ee36cb06f). I added this property to case modeler in order to resolve the conflicts with master branch. I don't really know what this property is used for. 

> * What about i18n internationalization? I have seen in the code, there is also CaseManagementDefinitionSetConstants-ja.properties file, but not updated.

I changed this file before, but the change was reverted by "kiereleaseuser". I guess the i18n team has their own plan to translate the properties. So I didn't add anything to this file this time.

**Comment by xiezhang7** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/variables/ProcessData.java [hunk 1]
```diff
@@ -32,7 +31,7 @@
 @Bindable
 @PropertySet
 @FormDefinition
-public class ProcessData implements BPMNPropertySet {
+public class ProcessData implements BaseProcessData {
```
[the remark this answers is not in the document]
Comment (xiezhang7) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/variables/ProcessData.java]: Fixed. Thanks.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/variables/ProcessVariables.java [hunk 2]
```diff
@@ -32,7 +31,7 @@
 @Bindable
 @Property
 @FieldDefinition(i18nMode = I18nMode.OVERRIDE_I18N_KEY)
-public class ProcessVariables implements BPMNProperty {
+public class ProcessVariables implements BaseProcessVariables {
```
[the remark this answers is not in the document]
Comment (xiezhang7) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/variables/ProcessVariables.java]: Fixed. Thanks.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-backend/src/test/java/org/kie/workbench/common/stunner/cm/backend/converters/tostunner/properties/CaseManagementActivityPropertyReaderTest.java [hunk 3]
```diff
@@ -78,4 +79,34 @@ public void testIsCase_false() throws Exception {
...
+
+        tested = new CaseManagementActivityPropertyReader(callActivity,
+                                                          definitionResolver.getDiagram(),
+                                                          definitionResolver);
+
+        assertTrue(tested.isAdHocAutostart());
+    }
+
+    @Test
+    public void testIisAdHocAutostart_false() throws Exception {
```
[the remark this answers is not in the document]
Comment (xiezhang7) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-backend/src/test/java/org/kie/workbench/common/stunner/cm/backend/converters/tostunner/properties/CaseManagementActivityPropertyReaderTest.java]: Fixed. Thanks.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-backend/src/test/java/org/kie/workbench/common/stunner/cm/backend/converters/tostunner/properties/CaseManagementAdHocSubProcessPropertyReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        tested = new CaseManagementAdHocSubProcessPropertyReader(adHocSubProcess,
+                                                                 definitionResolver.getDiagram(),
+                                                                 definitionResolver);
+
+        assertTrue(tested.isAdHocAutostart());
+    }
+
+    @Test
+    public void testIisAdHocAutostart_false() throws Exception {
```
[the remark this answers is not in the document]
Comment (xiezhang7) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-backend/src/test/java/org/kie/workbench/common/stunner/cm/backend/converters/tostunner/properties/CaseManagementAdHocSubProcessPropertyReaderTest.java]: Fixed. Thanks.

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Pull request merged**